src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1666,6 +1671,49 @@ void CodeGen::genAvxOrAvx2Intrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_AVX512F_MoveMaskSpecial:
+        {
+            op1Reg            = op1->GetRegNum();
+            regNumber maskReg = node->ExtractTempReg(RBM_ALLMASK);
+
+            instruction maskIns;
+            instruction kmovIns;
+
+            switch (baseType)
+            {
+                case TYP_BYTE:
+                case TYP_UBYTE:
+                    maskIns = INS_vpmovb2m;
+                    kmovIns = INS_kmovq;
+                    break;
+                case TYP_SHORT:
+                case TYP_USHORT:
+                    maskIns = INS_vpmovw2m;
+                    kmovIns = INS_kmovd;
+                    break;
+                case TYP_INT:
+                case TYP_UINT:
+                case TYP_FLOAT:
+                    maskIns = INS_vpmovd2m;
+                    kmovIns = INS_kmovw;
+                    break;
+                case TYP_DOUBLE:
+                case TYP_LONG:
+                case TYP_ULONG:
+                    maskIns = INS_vpmovq2m;
+                    kmovIns = INS_kmovb;
+                    break;
+                default:
+                    unreached();
+            }
+
+            assert(emitter::isMaskReg(maskReg));
```

Comment:
For now, should we update the assert to be:
```suggestion
            assert(emitter::isMaskReg(maskReg) && (maskReg == K1));
```